test/core/test_html_structure.rb

```diff
@@ -115,6 +115,30 @@ def test_render_with_custom_shortcut_and_additional_attrs
                 source, shortcut: {'^' => {tag: 'script', attr: 'data-binding', additional_attrs: { type: "application/json" }}}
   end
 
+  def test_render_with_custom_lambda_shortcut
+    source = %q{
+~foo Hello
+}
+    assert_html '<div class="styled-foo">Hello</div>',
+                source, shortcut: {'~' => {attr: ->(v) {["class", "\"styled-#{v}\""]}}}
+  end
+
+  def test_render_with_custom_lambda_shortcut_and_multiple_values
+    source = %q{
+~foo~bar Hello
+}
+    assert_html '<div class="styled-foo styled-bar">Hello</div>',
+                source, shortcut: {'~' => {attr: ->(v) {["class", "\"styled-#{v}\""]}}}
+  end
+
+#   def test_render_with_custom_lambda_shortcut_and_existing_class
+#     source = %q{
+# ~foo.baz Hello
```

Comment:
I have to revisit this and figure out how to support passing both the lambda shortcut and the regular `.` shortcut for `class`.